### ISSUE TITLE
translator.py breaks if a class with no base class is found inside a translator header

### ISSUE THREAD

**Issue by fjtc** [doxygen/doxygen#9882]
**Describe the bug**

`translator.py` breaks if a class with no base class is found inside a translator header.  On the other hand, if the sole class inside the translator has any base class, the report is generated as expected regardless of the fact that the class is not a Translator.

**Expected behavior**

Since it is a tool designed for the maintainers, I see no issues if this tool terminates with an error. The only thing that I think would be nice to do is to at least report the name of the affected class because the way `translator.py` terminates gives no hints about the cause of the sudden stop.

**To Reproduce**

Add a file `src/translator_zz.h` with the following contents:

```c++
#ifndef TRANSLATOR_ZZ_H
#define TRANSLATOR_ZZ_H

class Dummy {
};
#endif
```
and run the script.

The error vanishes if you replace the line `class Dummy {` with `class Dummy : QCString {` reporting that `Dummy` is missing all methods required by the Translator subclasses.

**Version**

Tested on 1.9.7 but it probably happens with previous versions as well.

**Stack trace**

```
Traceback (most recent call last):
  File "/home/<>/src/doxygen/doc/translator.py", line 2030, in <module>
    trMan = TrManager()
  File "/home/<>/src/doxygen/doc/translator.py", line 1293, in __init__
    self.__build()
  File "/home/<>/src/doxygen/doc/translator.py", line 1337, in __build
    tr.processing()
  File "/home/<>/src/doxygen/doc/translator.py", line 1059, in processing
    self.__collectPublicMethodPrototypes(tokenIterator)
  File "/home/<>/src/doxygen/doc/translator.py", line 734, in __collectPublicMethodPrototypes
    assert(self.baseClassId != None)
AssertionError
```

**Additional context**

If we replace the line 734:

```
assert(self.baseClassId != None)
```

by:

```
assert self.baseClassId != None, 'Class ' + self.classId + ' from the file ' + self.fname + ' does not have a base class.'
```

A more meaningful message will be displayed instead.

**Comment by albert-github** [comment]
A proposed patch has been implemented as part of pull request #9881

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
